### PR TITLE
Fixed: When we zoom the image, we will be able to slide to next or pr…

### LINKED ISSUE #1451: Switch to the next photo while panning at the edge of the current photo
If we've already reached the edge of the current photo in the zoomed out state, panning further should switch over to the next photo.

Of the top of my head, we could attach a different `GestureDetector` and use the `scale` attribute of the `PhotoView` to compute this:

https://github.com/ente-io/photos-app/blob/177b1bf3bb42589ccc0a73183a648b3f4023c8d6/lib/ui/viewer/file/zoomable_image.dart#L117

### PR DESCRIPTION
**Pull request by i-aiymen**
## Description

Functionality to switch the image to left or right, even if the images is zoomed

Closes #1451

### CODE REVIEW

**Comment by i-aiymen** [comment]
@vishnukvmd Just check it out

File: lib/ui/viewer/file/zoomable_image.dart [hunk 1]
```diff
@@ -89,21 +90,17 @@ class _ZoomableImageState extends State<ZoomableImage>
...
-              ? PhotoViewComputedScale.covered
-              : PhotoViewComputedScale.contained,
-          gaplessPlayback: true,
-          heroAttributes: PhotoViewHeroAttributes(
-            tag: widget.tagPrefix! + _photo.tag,
-          ),
-          backgroundDecoration: widget.backgroundDecoration as BoxDecoration?,
-        ),
+      content = PhotoViewGallery.builder(
+        builder: (context, index) {
```
Comment (vishnukvmd) [inline lib/ui/viewer/file/zoomable_image.dart:94]: All of the missing parameters (axis, controller, scaleStateChangedCallback, ...) are necessary.

Comment (i-aiymen) [reply lib/ui/viewer/file/zoomable_image.dart:94]: Actually, it is needed for photoView widget, for PhotoViewGallery.builder, we don't need all that. I have tested by adding that and removing it. It is of no difference @vishnukvmd

Comment (vishnukvmd) [reply lib/ui/viewer/file/zoomable_image.dart:94]: All of these parameters are either used upstream, when they're injected from a widget above in the render tree (for eg. `minScale` and `backgroundDecoration`), or used within this same widget (for eg. `gaplessPlayback` and `controller` ensures that the thumbnail is replaced by the full-res image seamlessly).

Looking into their usages will give you a better idea.

Comment (i-aiymen) [reply lib/ui/viewer/file/zoomable_image.dart:94]: ok

**Comment by i-aiymen** [comment]
@vishnukvmd Updated

**Review by vishnukvmd — APPROVED** [review]

**Pull request merged**